forward/fwd.go

```diff
@@ -278,12 +278,30 @@ func (f *websocketForwarder) serveHTTP(w http.ResponseWriter, req *http.Request,
 }
 
 // copyRequest makes a copy of the specified request.
-func (f *websocketForwarder) copyRequest(req *http.Request) (outReq *http.Request) {
+func (f *websocketForwarder) copyRequest(req *http.Request, u *url.URL) (outReq *http.Request) {
 	outReq = new(http.Request)
-	*outReq = *req
+	*outReq = *req // includes shallow copies of maps, but we handle this below
+
 	outReq.URL = utils.CopyURL(req.URL)
-	outReq.URL.Scheme = req.URL.Scheme
-	outReq.URL.Host = req.URL.Host
+	outReq.URL.Scheme = u.Scheme
+	outReq.URL.Host = u.Host
+	outReq.URL.Opaque = req.RequestURI
+	// raw query is already included in RequestURI, so ignore it to avoid dupes
+	outReq.URL.RawQuery = ""
+
+	outReq.Proto = "HTTP/1.1"
+	outReq.ProtoMajor = 1
+	outReq.ProtoMinor = 1
```

Comment:
Okay, thanks! I'll try to get a test case together just to be sure and then I'll follow up.